paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -436,13 +453,14 @@ std::unique_ptr<ir::Graph> MultiDevSSAGraphBuilder::ApplyImpl(
       }
     }
   }
-
   bool use_gpu = false;
 #ifdef PADDLE_WITH_CUDA
   use_gpu = nccl_ctxs_ != nullptr;
 #endif
 
-  if (use_gpu && strategy_.reduce_ == BuildStrategy::ReduceStrategy::kReduce) {
+  if ((use_gpu &&
```

Comment:
Add comments for this condition? It may be complicated for readers to understand.

Reply:
Sure, done.